ruleset/rootcheck/db/rootkit_trojans.txt

```diff
@@ -40,7 +40,7 @@ sudo        !satori|vejeta|conf\.inv!
 crond       !/dev/[^nt]|bash!
 gpm         !bash|mingetty!
 ifconfig    !bash|^/bin/sh|/dev/tux|session.null|/dev/[^cludisopt]!
-diff        !bash|^/bin/sh|file\.h|proc\.h|/dev/[^nf]ull|^/bin/.*sh!
+diff        !bash|^/bin/sh|file\.h|proc\.h|/dev/[^nf]|^/bin/.*sh!
```

Comment:
```suggestion
diff        !bash|^/bin/sh|file\.h|proc\.h|/dev/[^nf][^u]|^/bin/.*sh!
```

@brownowski as you left some notes in #25634 any opinion on this alternative?

It would still not catch `/dev/fuse` but would at least prevent an additional false negative against e.g. `/dev/fd` and `/dev/net` AFAICT.

Reply:
I believe something like the below would minimise corner cases.

/dev/[^nf]|/dev/[nf][^u]|/dev/[nf]u[^l]|/dev/[nf]ul[^l]

This catches:
All devices but devices starting with n or f
Then all devices starting with n or f, but not followed by u
Then all devices starting with n or f, then u, but not followed by l
Then all devices starting with n or f, then u, then l, but not followed by l